refactor/src/Reservoir.py

```diff
@@ -7,55 +9,106 @@ class Reservoir(CageTemplate):
     """
     The reservoir for sea lice, essentially modelled as a sea cage.
     """
+
     def __init__(self, cfg):
-        """
-        Create a cage on a farm
-        :param farm: the farm this cage is attached to
-        :param label: the label (id) of the cage within the farm
-        :param nplankt: TODO ???
+        """Create a reservoir
+
+        :param cfg: Simulation configuration and parameters
+        :type cfg: src.Config.Config
         """
 
         super().__init__(cfg)
 
         self.date = cfg.start_date
         self.num_fish = cfg.reservoir_num_fish
 
-        # this is by no means optimal: I want to distribute nplankt lice across
-        # each stage at random. (the method I use here is to create a random list
-        # of stages and add up the number for each.
-        lice_stage = np.random.choice(range(2, 7), cfg.reservoir_num_lice)
-        num_lice_in_stage = np.array([sum(lice_stage == i) for i in range(1, 7)])
-        self.lice_population = {'L1': num_lice_in_stage[0],
-                                'L2': num_lice_in_stage[1],
-                                'L3': num_lice_in_stage[2],
-                                'L4': num_lice_in_stage[3],
-                                'L5f': num_lice_in_stage[4],
-                                'L5m': num_lice_in_stage[5]}
+        # define life stage labels
+        life_stages = ["L1", "L2", "L3", "L4", "L5f", "L5m"]
+        n = len(life_stages)
+
+        # set probability of each life stage - in this case,
+        # the same of all lifestages
+        life_stages_prob = np.full(n, 1/n)
+
+        # construct rng generator
+        rng = np.random.default_rng()
+
+        # generate distribution based on life stage probs that sums
+        # up to total number of initial lice in reservoir
+        dist = rng.multinomial(cfg.reservoir_num_lice,
+                               life_stages_prob,
+                               size=1)[0]
+
+        # construct the population dict by assigning the numbers to labels
+        self.lice_population = {life_stages[ix]: dist[ix] for ix in range(n)}
 
     def to_csv(self):
         """
-        Save the contents of this cage as a CSV string for writing to a file later.
+        Save the contents of this cage as a CSV string
+        for writing to a file later.
         """
-        return f"reservoir, {self.num_fish}, {self.lice_population['L1']}, \
-                {self.lice_population['L2']}, {self.lice_population['L3']}, \
-                {self.lice_population['L4']}, {self.lice_population['L5f']}, \
-                {self.lice_population['L5m']}, {sum(self.lice_population.values())}"
 
-    def update(self, step_size, farms):
-        """
-        Update the reservoir at the current time step.
-        :return:
+        return f"reservoir, {self.num_fish}, \
+                {self.lice_population['L1']}, \
```

Comment:
Given that dictionary keys are sorted by key insertion order, and that `dict_values` is iterable, I think `", ".join(str(x) for x in self.lice_population.values())` would work? Sure it may be less readable this way...